src/light.js

```diff
@@ -57,20 +67,23 @@ const Type_Directional = 0;
 const Type_Point = 1;
 const Type_Spot = 2;
 
-class UniformLight
+class UniformLight extends UniformStruct
 {
     constructor()
     {
-        this.direction = jsToGl([0, 0, -1]);
+        super();
+
+        const defaultDirection = vec3.fromValues(0.7399, 0.6428, 0.1983);
+        this.direction = defaultDirection;
         this.range = -1.0;
```

Comment:
This `range` is the one that will be overwritten with a `undefined` as it is now.